internal/testrunner/runners/system/servicedeployer/compose.go

```diff
@@ -78,8 +78,14 @@ func (d *DockerComposeServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedSer
 			d.sv.Env...),
 		ExtraArgs: []string{"--build", "-d"},
 	}
-	if err := p.Up(opts); err != nil {
-		return nil, errors.Wrap(err, "could not boot up service using docker compose")
+	err = p.Up(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not boot up service using Docker Compose")
+	}
+
+	err = p.WaitForHealthy(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "service is unhealthy")
```

Comment:
Could we add as tests an scenario with system tests that would fail consistently if they are run without waiting for healthy services?

Reply:
Hm... I'm wondering how we can approach this as in most cases metrics will *eventually* flow. We would need a way to collect them immediately.